### PR TITLE
add github-actions workflow

### PR DESCRIPTION
**Pull request by dennybaa**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+COPY .versions /
+RUN apk add --no-cache ca-certificates git bash curl jq sed coreutils tar && \
+  . /.versions && \
+  ## Install kubectl of a given version \
+    ( cd /usr/local/bin && curl --retry 3 -sSLO \
+        "https://storage.googleapis.com/kubernetes-release/release/${KUBECTL_VERSION}/bin/linux/amd64/kubectl" && \
+      chmod 755 kubectl ) && \
+  ## Install helm \
+    ( cd /tmp && file="helm-${HELM_VERSION}-linux-amd64.tar.gz" && curl -sSLO https://get.helm.sh/$file && \
+       printf "${HELM_SHA}  ${file}" && tar zxf ${file} && mv linux-amd64/helm /usr/local/bin/ ) && \
```
Comment (voron) [inline Dockerfile]: Is it expected to don't  check checksum here?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,46 @@
+ARG FLAVOR_IMAGE
+FROM ${FLAVOR_IMAGE}
+ARG KUBECTL_VERSION=v1.21.1
+
+COPY .versions /
+RUN apk add --no-cache ca-certificates git bash curl jq sed coreutils tar && \
+  . /.versions && \
+  ## Install kubectl of a given version \
+    ( cd /usr/local/bin && curl --retry 3 -sSLO \
+        "https://storage.googleapis.com/kubernetes-release/release/${KUBECTL_VERSION}/bin/linux/amd64/kubectl" && \
```
Comment (voron) [inline Dockerfile:11]: WDYT on specifying explicitly in a comment something like "no checksum check as kubectl version is dynamic from argument" ?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      mkdir /dysnix && adduser kubectl -u 1001 -D -h /dysnix/kubectl; \
+    fi
+
+USER ${USER}
+WORKDIR /dysnix/kubectl
+
+## Install plugins (already as the specified user)
+RUN helm plugin install https://github.com/databus23/helm-diff && \
+    helm plugin install https://github.com/futuresimple/helm-secrets && \
+    helm plugin install https://github.com/hypnoglow/helm-s3.git && \
```
Comment (voron) [inline Dockerfile:42]: WDYT on helm-gcs?

File: .github/workflows/build-and-push.yaml [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+      - id: set
+        run: |
+          # Computed org/repo name, docker- prefix is stripped off
+          export DEFAULT_IMAGE=$(echo -n '${{ github.event.organization.login }}/'; echo '${{ github.event.repository.name }}' | sed 's/^docker-//' | tr '[:upper:]' '[:lower:]')
+          echo "::set-output name=image::${SET_NAME:-$DEFAULT_IMAGE}"
+      - id: version
+        if: startsWith(github.ref, 'refs/tags/')
+        run: |
+          # Computed version (pick tag's stripped semver, i.e. only vX.Y.Z)
+          KUBECTL_VERSION=$(echo '${{ github.ref }}' | sed -E -e 's/refs\/tags\///' -e 's/(v[0-9]+.[0-9]+.[0-9]+).*/\1/')
```
Comment (voron) [inline .github/workflows/build-and-push.yaml:29]: I've used to `;` like 
```
sed -E -e 's/refs\/tags\///;s/(v[0-9]+.[0-9]+.[0-9]+).*/\1/'
```
Didn't know about multiple `-e`, it's a more cleaner way!

File: .github/workflows/build-and-push.yaml [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+          repository: ${{ github.repository }}
+      -
+        name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v3
+        with:
+          images: ${{ needs.pre.outputs.image }}
+          flavor: |
+            latest=${{ steps.latestTag.outputs.older == 'false' }}
+            suffix=${{ matrix.suffix }}
```
Comment (voron) [inline .github/workflows/build-and-push.yaml:67]: Does it work fine w/ non-defined suffix w/ default flavor?

File: .github/workflows/build-and-push.yaml [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+      -
+        name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+      -
+        name: Cache Docker layers
+        uses: actions/cache@v2
+        with:
+          path: /tmp/.buildx-cache
+          key: ${{ runner.os }}-buildx-${{ matrix.flavor }}-${{ github.sha }}
+          restore-keys: |
```
Comment (voron) [inline .github/workflows/build-and-push.yaml:80]: do we need a multiline here just to easily compare ( w/ eyes, I mean ) the value with `key` value ?

**Review by voron — APPROVED** [review]
LGTM, need to try it!
Why do we need an empty file named `f` ?

File: Dockerfile (hunk 1 above)
Comment (dennybaa) [reply Dockerfile]: yup kubectl, we don't expect to checksum ) We don't know it, since the version is fetched based on the tag released.

Comment (voron) [reply Dockerfile]: I've asked about helm, not a kubectl. helm version is hardcoded in `.versions`

**Pull request merged**